packages/artifact/src/internal-download-http-client.ts

```diff
@@ -12,122 +12,174 @@ import {
- */
-export async function getContainerItems(
-  artifactName: string,
-  containerUrl: string
-): Promise<QueryArtifactResponse> {
-  // The itemPath search parameter controls which containers will be returned
-  const resourceUrl = new URL(containerUrl)
-  resourceUrl.searchParams.append('itemPath', artifactName)
+  /**
+   * Gets a list of all artifacts that are in a specific container
+   */
+  async listArtifacts(): Promise<ListArtifactsResponse> {
+    const artifactUrl = getArtifactUrl()
+    this.downloadHttpManager.createClients(1)
+    const client = this.downloadHttpManager.getClient(0)
+    const requestOptions = getRequestOptions('application/json')
 
-  const client = createHttpClient()
-  const rawResponse = await client.get(resourceUrl.toString())
-  const body: string = await rawResponse.readBody()
-  if (isSuccessStatusCode(rawResponse.message.statusCode) && body) {
-    return JSON.parse(body)
+    const rawResponse = await client.get(artifactUrl, requestOptions)
+    const body: string = await rawResponse.readBody()
+    if (isSuccessStatusCode(rawResponse.message.statusCode) && body) {
+      return JSON.parse(body)
+    }
+    // eslint-disable-next-line no-console
+    console.log(rawResponse)
+    throw new Error(`Unable to list artifacts for the run`)
   }
-  // eslint-disable-next-line no-console
-  console.log(rawResponse)
-  throw new Error(`Unable to get ContainersItems from ${resourceUrl}`)
-}
 
-/**
- * Concurrently downloads all the files that are part of an artifact
- * @param downloadItems information about what items to download and where to save them
- */
-export async function downloadSingleArtifact(
-  downloadItems: DownloadItem[]
-): Promise<void> {
-  const DOWNLOAD_CONCURRENCY = getDownloadFileConcurrency()
-  // Limit the number of files downloaded at a single time
-  const parallelDownloads = [...new Array(DOWNLOAD_CONCURRENCY).keys()]
-  const client = createHttpClient()
-  let downloadedFiles = 0
-  await Promise.all(
-    parallelDownloads.map(async () => {
-      while (downloadedFiles < downloadItems.length) {
-        const currentFileToDownload = downloadItems[downloadedFiles]
-        downloadedFiles += 1
-        await downloadIndividualFile(
-          client,
-          currentFileToDownload.sourceLocation,
-          currentFileToDownload.targetPath
-        )
-      }
-    })
-  )
-}
+  /**
+   * Fetches a set of container items that describe the contents of an artifact
+   * @param artifactName the name of the artifact
+   * @param containerUrl the artifact container URL for the run
+   */
+  async getContainerItems(
+    artifactName: string,
+    containerUrl: string
+  ): Promise<QueryArtifactResponse> {
+    // the itemPath search parameter controls which containers will be returned
+    const resourceUrl = new URL(containerUrl)
+    resourceUrl.searchParams.append('itemPath', artifactName)
+    this.downloadHttpManager.createClients(1)
```

Comment:
Same here.  If the intention is just to have a single http client, just initiate a single http client.